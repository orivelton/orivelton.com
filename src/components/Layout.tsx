import { MetaProps } from '../types/layout'
import Head from './Head'
import Navigation from './Navigation'
import ThemeSwitch from './ThemeSwitch'
import { AiFillGithub, AiFillLinkedin } from 'react-icons/ai'

type LayoutProps = {
  children: React.ReactNode
  customMeta?: MetaProps
}

export const WEBSITE_HOST_URL = 'https://orivelton.com'

const Layout = ({ children, customMeta }: LayoutProps): JSX.Element => {
  return (
    <>
      <Head customMeta={customMeta} />
      <header>
        <div className="max-w-5xl px-8 mx-auto">
          <div className="flex items-center justify-between py-6">
            <Navigation />
            <ThemeSwitch />
          </div>
          <hr />
        </div>
      </header>
      <main>
        <div className="max-w-5xl px-8 py-4 mx-auto">{children}</div>
      </main>
      <footer className="py-8">
        <div className="max-w-5xl px-8 mx-auto ">
          <p className="flex flex-row justify-center">
            Orivelton CÃ©sar - { new Date().getFullYear() }
          </p>
          <ul className="flex flex-row justify-center">
            <li>
              <a href="https://github.com/orivelton" title="Go to my profile on Github" rel="noopener noreferrer nofollow" target="_blank">
                <AiFillGithub />
              </a>
            </li>
            <li className="ml-4">
              <a href="https://www.linkedin.com/in/orivelton/" title="Go to my profile on LikedIn" rel="noopener noreferrer nofollow" target="_blank">
                <AiFillLinkedin />
              </a>
            </li>
          </ul>

        </div>
      </footer>
    </>
  )
}

export default Layout
