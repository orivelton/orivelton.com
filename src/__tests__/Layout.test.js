import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../components/Layout';

describe('src/components/Layout.js', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Layout />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
});
