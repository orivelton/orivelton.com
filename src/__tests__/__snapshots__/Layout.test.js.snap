// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Layout.js renders correctly 1`] = `
Array [
  <header>
    <nav
      className="nav"
    >
      <div
        className="nav__menu"
      >
        <a
          className="nav__link nav__link--logo"
          href="/"
          title="Go to Home"
        >
          <img
            alt="Logo Orivelton Cesar"
            className="logo"
            src="test-file-stub"
          />
        </a>
        <ul
          className="nav__group"
        >
          <li>
            <a
              className="nav__link"
              href="/blog"
              title="Got to Blog page"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              className="nav__link nav__link--no-padding-right"
              href="/contact"
              title="Got to Contact page"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
      <hr />
    </nav>
  </header>,
  <div
    className="content"
  >
    <main />
  </div>,
  <footer>
    Â© 
     
    <a
      href="/"
      style={
        Object {
          "boxShadow": "none",
          "color": "inherit",
        }
      }
    >
      Orivelton Cesar
    </a>
     
    2020
  </footer>,
]
`;
